test/codec.js

```diff
@@ -990,4 +996,117 @@ describe('codec', function() {
       });
     });
   });
+
+  describe('createTypeObject', function() {
+    var TYPES = [
+      'unspecified',
+      'bool',
+      'int64',
+      'float64',
+      'timestamp',
+      'date',
+      'string',
+      'bytes',
+    ];
+
+    it('should convert the type to its int value', function() {
+      TYPES.forEach(function(typeName, i) {
+        var type = codec.createTypeObject(typeName);
+
+        assert.deepEqual(type, {code: i});
+      });
+    });
+
+    it('should default to unspecified for unknown types', function() {
+      var type = codec.createTypeObject('unicorn');
+
+      assert.deepEqual(type, {code: 0});
+    });
+
+    it('should set the arrayElementType', function() {
+      var type = codec.createTypeObject({
+        type: 'array',
+        child: 'bool',
+      });
+
+      assert.deepEqual(type, {
+        code: 8,
```

Comment:
Can we do `TYPES.indexOf('array')` here? (Same for other places in the tests that we use the magic numbers)

Reply:
Done!